moveit_ros/planning/kinematics_plugin_loader/include/moveit/kinematics_plugin_loader/kinematics_plugin_loader.h

```diff
@@ -57,7 +60,23 @@ class KinematicsPluginLoader
   KinematicsPluginLoader(const rclcpp::Node::SharedPtr& node,
                          const std::string& robot_description = "robot_description",
                          double default_search_resolution = 0.0)
-    : node_(node), robot_description_(robot_description), default_search_resolution_(default_search_resolution)
+    : node_interface_(node->get_node_base_interface())
+    , parameters_interface_(node->get_node_parameters_interface())
+    , topics_interface_(node->get_node_topics_interface())
+    , robot_description_(robot_description)
+    , default_search_resolution_(default_search_resolution)
+  {
+  }
+
+  KinematicsPluginLoader(const NodeInterfaceSharedPtr& node_interface, const TopicsInterfaceSharedPtr& topics_interface,
```

Comment:
Where should that struct be located? Since these changes are across multiple packages, would it make sense to create a utility package in /moveit_ros that define a wrapper struct?

Reply:
That's the question... We've discussed abstracting from the RCLCPP interfaces in the past, and came up with some DI approach for increased testability. https://github.com/ros-planning/moveit2/pull/562 and https://github.com/ros-planning/moveit2/pull/569 are examples for classes that provide ROS features without direct API usage. However, these interfaces were very specific to their classes. It would be great, if we could come up with a more generic approach for the node interfaces, which technically could be used all over the codebase. I'm not saying that this needs to be done with this PR, but if we could reduce it to one type, refactoring would be much easier in the future.

> Since these changes are across multiple packages, would it make sense to create a utility package in /moveit_ros that define a wrapper struct?

Yes, that would make sense to me in this case.

Tagging @tylerjw @griswaldbrooks for their opinion.

Reply:
I made a package called moveit_node_interface that wraps the lifecycle and regular node.